lib/hcloud/server.rb

```diff
@@ -37,10 +37,17 @@ def create_image(description: nil, type: nil)
     end
 
     def rebuild(image:)
+      raise Hcloud::Error::InvalidInput, 'no image given' if image.blank?
+
       prepare_request('actions/rebuild', j: { image: image }) { |j| j[:root_password] }
     end
 
     def change_type(server_type:, upgrade_disk: nil)
+      raise Hcloud::Error::InvalidInput, 'no server_type given' if server_type.blank?
+      # According to API docs, upgrade_disk is a required field. The docs do not
+      # mention that either true or false is a default.
+      raise Hcloud::Error::InvalidInput, 'no upgrade_disk given' if upgrade_disk.nil?
```

Comment:
In the frontend upgrade_disk defaults to false, because it's the "safe" way. You can always upscale the disk later if you need to, but you can never(!) downscale so true should never be the default. I wonder if we want to "copy" the default = false from the frontend because it's a safe choice but since this is an API client it's probably reasonable that the user should decide himself what he want to do here.

tl;dr - I would either set the default to false or remove the comment because it's "intended" that there is no default.